python/tests/test_list_definition.py

```diff
@@ -0,0 +1,60 @@
+import unittest
+
+from hyperon import *
+from common import MeTTa
+
+class ListDefinitionTest(unittest.TestCase):
+
+    # As per issue
+    #
+    # https://github.com/trueagi-io/hyperon-experimental/issues/104
+    #
+    # Test if adding type declaration to List data structure does
+    # not interfere with executing functions operating on List.
+    def test_list_definition(self):
+        metta = MeTTa()
+        metta.add_parse('''
+                ;; Define conditional
+                (= (if True $x $y) $x)
+                (= (if False $x $y) $y)
+
+                ;; Declaration of List data structure
+                ;; [TODO: uncomment the 3 lines below once the bug is fixed]
+                ;; (: List (-> $a Type))
+                ;; (: Nil (List $a))
+                ;; (: Cons (-> $a (List $a) (List $a)))
+
+                ;; Insert an element in a presumably sorted list
+                (= (insert $x Nil) (Cons $x Nil))
+                (= (insert $x (Cons $head $tail))
+                  (if (< $x $head)
+                      (Cons $x (Cons $head (Cons $tail)))
```

Comment:
Nil, small typo here:
```suggestion
                      (Cons $x (Cons $head $tail))
```

Reply:
Thanks @vsbogd !